File: backend-mobile/src/main/java/com/nimalsha/model/Nutritionist.java
package com.nimalsha.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Nutritionist {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private Long userid;

    private String fullName; // Nutritionist's full name
    private String email;    // Nutritionist's email address
    private String specializations; // Specializations, can be a comma-separated string
    private int experience;  // Number of years of experience
    private String qualifications;

    @Lob
    private byte[] documents; // Supporting documents in binary format

    
}

File: fiteatsmobile/lib/models/nutritionist.dart
// nutritionist.dart (Model)

class Nutritionist {
  final int id;
  final String name;
  final String photo;
  final String qualifications;
  final List<String> specializations;

  Nutritionist({
    required this.id,
    required this.name,
    required this.photo,
    required this.qualifications,
    required this.specializations,
  });

  factory Nutritionist.fromJson(Map<String, dynamic> json) {
    return Nutritionist(
      id: json['id'],
      name: json['name'],
      photo: json['photo'],
      qualifications: json['qualifications'],
      specializations: List<String>.from(json['specializations']),
    );
  }
}
